geolocation.py

```diff
@@ -6,8 +6,60 @@
     CODE_COUNTRY_DICT,
     STATE_ABBREV,
     STATE_NAMES,
+    CITY_COUNTRY_STRINGS,
 )
 
+def levenshteinDistance(s1, s2):
+    """Calculated Levenstein edit distance between two arbitary strings
+
+    Args:
+        s1: an arbitrary string
+        s2: a second arbitrary string
+
+    Return:
+        str: an integer value of how distant the two strings are
+    """
+    # special cities and countries
+    if len(s1) > len(s2):
+        s1, s2 = s2, s1
+
+    distances = range(len(s1) + 1)
+    for i2, c2 in enumerate(s2):
+        distances_ = [i2+1]
+        for i1, c1 in enumerate(s1):
+            if c1 == c2:
+                distances_.append(distances[i1])
+            else:
+                distances_.append(1 + min((distances[i1], distances[i1 + 1], distances_[-1])))
+        distances = distances_
+    return distances[-1]
+
+def edit_distance_to_world(location):
+    """Uses Levenstein distance to try to approximate a match to city+country strings
+
+    Args:
+        location_string: a text containing user-supplied location
+
+    Return:
+        str: a country
+    """
+    # special cities and countries
+    special = ["US", "USA", "U.S.A", "U.S.", 'San Francisco']
```

Comment:
@kdobolyi, what do you think about a variable name longer than special? What about special_locations?